linkedlist.py

```diff
@@ -19,29 +18,69 @@ def pop(self):
             self.head = self.head.next
             self._length -= 1
         else:
-            raise ValueError("Nothing")
+            raise ValueError("Empty List")
         return old_head.value
     
     def size(self):
         return self._length
                 
 
-    def search(self, value):
-        for val in self.values:
-            if val == value:
-                return value
-            
+    def search(self, val):
+        # make sure your searching the LinkedList and not the values
+        current = self.head
+        count = 0
+        if self.head:
+            while count < self._length:
+                if current.value == val:
+                    return current.value
+                current = current.next
+                count += 1
         return None
     
     def remove(self, node):
-        pass
-
+        current = self.head
+        previous = None
+        while current:
+            if current == node:
+                    if current == self.head:
+                        self.head = self.head.next
+                        self._length -= 1
+                        return
+                    previous.next = current.next
+                    self._length -= 1
+                    return
+            previous = current
+            current = current.next
+        raise ValueError("node not in LinkedList")
+    
+    def display(self):
+        statement = "("
+        current = self.head
+        while current:
+            statement += str(current.value) + ", "
+            current = current.next
+        return statement[:-2] + ")"
+    
+    def len(self):
+        return self._length
+    
+    def print(self):
+        return self.display()
             
+        
 
 
 class Node():
-    def __init__(self, value, next):
+    def __init__(self, value, next, previous):
         self.value = value
         self.next = next
+        self.previous = previous
```

Comment:
Could link to the Data Structure you are doing?  If this is a Regular linked list, there is no `previous`. The linking is one directional. If you are working on the doubly linked list the items are linked both directions.